x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/process/NativeAnalyticsProcessFactory.java

```diff
@@ -88,6 +88,11 @@ private void createNativeProcess(String jobId, AnalyticsProcessConfig analyticsP
         } catch (IOException e) {
             String msg = "Failed to launch data frame analytics process for job " + jobId;
             LOGGER.error(msg);
+            try {
```

Comment:
I wonder if we could catch the exception in `ProcessPipes.connectStreams`, close unused streams and rethrow. This way callers would not need to worry about this.

Reply:
Well that seems obvious now. I made the change